source/common/config/unified_mux/delta_subscription_state.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include "envoy/grpc/status.h"
+
+#include "source/common/common/assert.h"
+#include "source/common/common/logger.h"
+#include "source/common/config/api_version.h"
+#include "source/common/config/unified_mux/subscription_state.h"
+
+#include "absl/container/node_hash_map.h"
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Config {
+namespace UnifiedMux {
+
+// Tracks the state of a delta xDS-over-gRPC protocol session.
+class DeltaSubscriptionState
+    : public SubscriptionState<envoy::service::discovery::v3::DeltaDiscoveryResponse,
+                               envoy::service::discovery::v3::DeltaDiscoveryRequest> {
+public:
+  DeltaSubscriptionState(std::string type_url, UntypedConfigUpdateCallbacks& watch_map,
```

Comment:
Nit: can you check all the places you are passing strings as args and move them to refs/string_views?

Reply:
will do

Reply:
I checked string parameters in ctors. In ctor implementations, the initial copy is being moved all the way to the base class member initialization (here: https://github.com/envoyproxy/envoy/pull/16486/files#diff-807032428ceb00bb04de1be63707b2ae3dccaa00912341208fd4b8da55993b17R38).

Reply:
I guess this is OK, but it's a bit counter-intuitive to someone reading the interface. Usually prefer to pass by reference and then leave the copy internally.